svn diff modules/core/src/org/apache/axis2/deployment/WarBasedAxisConfigurator.java
Index: modules/core/src/org/apache/axis2/deployment/WarBasedAxisConfigurator.java
===================================================================
--- modules/core/src/org/apache/axis2/deployment/WarBasedAxisConfigurator.java  (revision 411677)
+++ modules/core/src/org/apache/axis2/deployment/WarBasedAxisConfigurator.java  (working copy)
@@ -30,6 +30,11 @@
 *
 */

+/**
+* Processes the init parameters for the AxisServlet.
+This allows the location of the axis2.xml and the module repository to be different from the default locations.
+The init parameters support alternate file, or URL values for both of these.
+*/
 public class WarBasedAxisConfigurator implements AxisConfigurator {

     private AxisConfiguration axisConfig;
@@ -37,39 +42,82 @@
     private ServletConfig config;
     private DeploymentEngine deploymentEngine;

+
+    /**
+    * The name of the init parameter (axis2.xml.path) that can be used to override the default location for the axis2.xml file. When both this init parameter, and the axis2.xml.url init parameters are not specified in the axis servlet init-parameter, the default location of ${app}/WEB-INF/conf/axis2.xml is used.
+    * This parameter takes precidence over the axis2.xml.url init parameter.
+    */
+    public static final String PARAM_AXIS2_XML_PATH = "axis2.xml.path";
+
+
+    /**
+    * The name of the init parameter (axis2.xml.url) that when specified indicates the axis2.xml should be loaded using the URL specified as the value of this init parameter. If the axis2.xml.path init parameter is present, this init parameter has no effect.
+    */
+    public static final String PARAM_AXIS2_XML_URL = "axis2.xml.url";
+
+
+    /**
+    * The name of the init parameter (axis2.repository.path) that when specified indicates the path to the
+    */
+    public static final String PARAM_AXIS2_REPOSITORY_PATH = "axis2.repository.path";
+
+
+    /**
+    * The name of the init parameter (axis2.repository.url) that when specified indicates the url to be used
+    */
+    public static final String PARAM_AXIS2_REPOSITORY_URL = "axis2.repository.url";
+
+
+    /**
+    * Default constructor for configurator.
+    * This determines the axis2.xml file to be used from the init parametes for the AxisServlet in the web.xml.
+    * The order of initialization is according the the following precidence:
+    * <ul>
+    * <li>If the parameter axis2.xml.path is present, the value is webapp relative path to be used as the location to the axis2.xml file.
+    * <li>Otherwise, if the parameter axis2.xml.url is present, the URL is used as the location to the axis2.xml file.
+    * <li>Otherwise, when both of the above init parameters are not present, file is attempted to be loaded from &lt;repo&gt;/WEB-INF/axis2.xml.
+    * <li> When none of the above could be found, the axis2.xml is loaded from the classpath resource, the value of DeploymenConstants.AXIS2_CONFIGURATION_RESOURCE.
+    * </ul>
+    *
+    * @param svconfig the ServletConfig object from the AxisServlet. This method is called from the init() of the AxisServlet.
+    */
     public WarBasedAxisConfigurator(ServletConfig svconfig) {
         try {
             this.config = svconfig;
             deploymentEngine = new DeploymentEngine();
-            String axis2xmlpath = config.getInitParameter("axis2.xml.path");
+            String axis2xmlpath = config.getInitParameter(PARAM_AXIS2_XML_PATH);
             String repository;
-            InputStream axis2Steram;
+            InputStream axis2Stream;
             if (axis2xmlpath != null) {
-                axis2Steram = new FileInputStream(axis2xmlpath);
-                axisConfig = deploymentEngine.populateAxisConfiguration(axis2Steram);
+                // when init parameter was present.
+                axis2Stream = new FileInputStream(axis2xmlpath);
+                axisConfig = deploymentEngine.populateAxisConfiguration(axis2Stream);
             } else {
-                String axisurl = config.getInitParameter("axis2.xml.url");
+                String axisurl = config.getInitParameter(PARAM_AXIS2_XML_URL);
                 if (axisurl != null) {
-                    axis2Steram = new URL(axisurl).openStream();
-                    deploymentEngine.populateAxisConfiguration(axis2Steram);
+                    axis2Stream = new URL(axisurl).openStream();
+                    axisConfig = deploymentEngine.populateAxisConfiguration(axis2Stream);
                 } else {
+                    // both the axis2.xml.path and axis2.xml.url init parameters were not present
                     try {
+                        // try to find the default /WEB-INF/conf/axis2.xml
                         repository = config.getServletContext().getRealPath("/WEB-INF");
-                        axis2Steram = new FileInputStream(repository + "/conf/axis2.xml");
-                        axisConfig = deploymentEngine.populateAxisConfiguration(axis2Steram);
+                        axis2Stream = new FileInputStream(repository + "/conf/axis2.xml");
+                        axisConfig = deploymentEngine.populateAxisConfiguration(axis2Stream);
                         setWebLocationProperty(config.getServletContext(), deploymentEngine);
                     } catch (Exception e) {
+                        // not there, use default configuration from class path resource.
                         ClassLoader cl = Thread.currentThread().getContextClassLoader();
-                        axis2Steram = cl.getResourceAsStream(DeploymentConstants.AXIS2_CONFIGURATION_RESOURCE);
-                        axisConfig = deploymentEngine.populateAxisConfiguration(axis2Steram);
+                        if (log.isWarnEnabled()) {
+                            log.warn("unable to find axis2.xml from init parameter axis2.xml.url or " + repository + "/conf/axis2.xml . Using default configuration: " + DeploymentConstants.AXIS2_CONFIGURATION_RESOURCE);
+                        }
+                        axis2Stream = cl.getResourceAsStream(DeploymentConstants.AXIS2_CONFIGURATION_RESOURCE);
+                        axisConfig = deploymentEngine.populateAxisConfiguration(axis2Stream);
                     }
                 }
             }
-            if(axisConfig == null) {
-                throw new RuntimeException("Unable to load axis2.xml configuration");
-            }
-            Parameter unableHttp = new Parameter("enableHTTP", "true");
-            axisConfig.addParameter(unableHttp);
+
+
         } catch (FileNotFoundException e) {
             log.info(e.getMessage());
         } catch (DeploymentException e) {
@@ -78,6 +126,13 @@
             log.info(e.getMessage());
         } catch (IOException e) {
             log.info(e.getMessage());
+        } finally {
+            try {
+                Parameter enableHttp = new Parameter("enableHTTP", "true");
+                axisConfig.addParameter(enableHttp);
+            } catch (AxisFault axisFault) {
+                log.info(axisFault.getMessage());
+            }
         }
     }

@@ -96,14 +151,27 @@
         depeng.setWebLocationString(weblocation.getAbsolutePath());
     }

+
+    /**
+    * Gets the axis configuration object by loading the repository.
+    * The order of initialization is according the the following precidence:
+    * <ul>
+    * <li>If the parameter axis2.repository.path is present, this folder is used as the location to the repository.
+    * <li>Otherwise, if the parameter axis2.repository.url is present, the URL is used as the location to the repository.
+    * <li>Otherwise, when both of the above init parameters are not present, the web applications WEB-INF folder is used as the folder for the repository.
+    * </ul>
+    *
+    * @return the instance of the AxisConfiguration object that reflects the repository according to the rules above.
+    * @throws AxisFault when an error occured in the initialization of the AxisConfiguration.
+    */
     public AxisConfiguration getAxisConfiguration() throws AxisFault {
         try {
             String repository;
-            String axis2repopath = config.getInitParameter("axis2.repository.path");
+            String axis2repopath = config.getInitParameter(PARAM_AXIS2_REPOSITORY_PATH);
             if (axis2repopath != null) {
                 deploymentEngine.loadRepository(axis2repopath);
             } else {
-                String axis2repourl = config.getInitParameter("axis2.repository.url");
+                String axis2repourl = config.getInitParameter(PARAM_AXIS2_REPOSITORY_URL);
                 if (axis2repourl != null) {
                     deploymentEngine.loadRepositoryFromURL(new URL(axis2repourl));
                 } else {
@@ -122,13 +190,19 @@
     }

     //to load services
+    /**
+    * Loads the services within the repository.
+    * When the axis2.repository.path init parameter was present, we just call loadServices() in the deployment engine.<br/>
+    * When the axis2.repository.url init parameter was present we load services from the respective URL value of the init parameter.<br/>
+    * Otherwise, try to load the services from the /WEB-INF folder within the web application.
+    */
     public void loadServices() {
         try {
-            String axis2repopath = config.getInitParameter("axis2.repository.path");
+            String axis2repopath = config.getInitParameter(PARAM_AXIS2_REPOSITORY_PATH);
             if (axis2repopath != null) {
                 deploymentEngine.loadServices();
             } else {
-                String axis2repourl = config.getInitParameter("axis2.repository.url");
+                String axis2repourl = config.getInitParameter(PARAM_AXIS2_REPOSITORY_URL);
                 if (axis2repourl != null) {
                     deploymentEngine.loadServicesFromUrl(new URL(axis2repourl));
                 } else {
@@ -140,7 +214,7 @@
                         log.info(e.getMessage());
                     }
                 }
-            }
+            } // else
         } catch (MalformedURLException e) {
             log.info(e.getMessage());
         }

